From 05aa60fc5d982b45e73adc674e1ef31b379c1f4a Mon Sep 17 00:00:00 2001
From: Mohammed Hassan <mohammed.hassan@jolla.com>
Date: Thu, 5 May 2016 03:39:57 +0300
Subject: [PATCH 4/4] jifmux: cope with missing EOI marker.

If EOI marker is not within the last 5 bytes of the jpeg stream then we will fail to find it.
Earlier versions of this patch scanned the whole stream, but it's simpler to just assume that
the rest of the data is scan data.
---
 gst/jpegformat/gstjifmux.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/gst/jpegformat/gstjifmux.c b/gst/jpegformat/gstjifmux.c
index e958f54d3..8f0dfd01e 100644
--- a/gst/jpegformat/gstjifmux.c
+++ b/gst/jpegformat/gstjifmux.c
@@ -312,6 +312,7 @@ gst_jif_mux_parse_image (GstJifMux * self, GstBuffer * buf)
 
     if (marker == SOS) {
       gint eoi_pos = -1;
+      gboolean eoi_found = FALSE;
       gint i;
 
       /* search the last 5 bytes for the EOI marker */
@@ -319,6 +320,7 @@ gst_jif_mux_parse_image (GstJifMux * self, GstBuffer * buf)
       for (i = 5; i >= 2; i--) {
         if (map.data[map.size - i] == 0xFF && map.data[map.size - i + 1] == EOI) {
           eoi_pos = map.size - i;
+          eoi_found = TRUE;
           break;
         }
       }
@@ -334,6 +336,10 @@ gst_jif_mux_parse_image (GstJifMux * self, GstBuffer * buf)
         goto error;
 
       GST_DEBUG_OBJECT (self, "scan data, size = %u", self->scan_size);
+
+      /* If we cannot find EOI marker, we assume the rest of the data is scan data */
+      if (!eoi_found)
+        goto done;
     }
 
     if (!gst_byte_reader_peek_uint8 (&reader, &marker))
-- 
2.25.1

